descartes_trajectory/include/descartes_trajectory/cartesian_interpolator.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2014, Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTESIAN_INTERPOLATOR_H_
+#define CARTESIAN_INTERPOLATOR_H_
+
+#include <descartes_trajectory/cart_trajectory_pt.h>
+#include <tf/transform_datatypes.h>
+
+namespace descartes_trajectory
+{
+
+class TrajectorySegment
```

Comment:
As this class and its children are not part of the public interface of the cartesian interpolater, would it make sense to put these in a separate header file?